config-model/src/main/java/com/yahoo/vespa/model/application/validation/change/RestartOnDeployForOnnxModelChangesValidator.java

```diff
@@ -60,7 +60,9 @@ public List<ConfigChangeAction> validate(VespaModel currentModel, VespaModel nex
     private Optional<String> modelChanged(OnnxModelCost.ModelInfo a, OnnxModelCost.ModelInfo b) {
         if (a.estimatedCost() != b.estimatedCost()) return Optional.of("estimated cost");
         if (a.hash() != b.hash()) return Optional.of("model hash");
-        if (! a.onnxModelOptions().equals(b.onnxModelOptions())) return Optional.of("model option(s)");
+        if (a.onnxModelOptions().isPresent() && b.onnxModelOptions().isPresent()
```

Comment:
So only if both exist, can they be different.. ?

Reply:
We only care about them if they are both present, really. Not having options is legacy (apps on newer versions always have them, will be simplified when all apps have upgraded to a newer version). But I will fix it